services/api/src/resources/environment/environment_redeploy.ts

```diff
@@ -38,67 +38,104 @@ _,
+        lc.ts as env_last_updated,
+        IF(ev.updated > lc.ts, 'deploy', 'no-deploy') as must_deploy,
+        'Environment' AS envvar_source,
+        3 as envvar_priority
+      FROM environment e
+      JOIN env_vars ev ON ev.environment = e.id
+      CROSS JOIN last_completed lc
+      WHERE e.id = ?
+        AND ev.name IS NOT NULL
+
+      UNION ALL
+
+      -- Project-scoped
+      SELECT
+        e.id, e.name,
+        ev.name AS envvar_name,
+        ev.updated AS envvar_updated,
+        lc.ts as env_last_updated,
+        IF(ev.updated > lc.ts, 'deploy', 'no-deploy') as must_deploy,
+        'Project' AS envvar_source,
+        2 as envvar_priority
+      FROM environment e
+      JOIN project p   ON p.id = e.project
+      JOIN env_vars ev ON ev.project = p.id
+      CROSS JOIN last_completed lc
+      WHERE e.id = ?
+        AND ev.name IS NOT NULL
+
+      UNION ALL
+
+      -- Organization-scoped
+      SELECT
+        e.id, e.name,
+        ev.name AS envvar_name,
+        ev.updated AS envvar_updated,
+        lc.ts as env_last_updated,
+        IF(ev.updated > lc.ts, 'deploy', 'no-deploy') as must_deploy,
+        'Organization' AS envvar_source,
+        1 as envvar_priority
+      FROM environment e
+      JOIN project p        ON p.id = e.project
+      JOIN organization o   ON o.id = p.organization
+      JOIN env_vars ev      ON ev.organization = o.id
+      CROSS JOIN last_completed lc
+      WHERE e.id = ?
+        AND ev.name IS NOT NULL
+
+    ) AS allenvs
+    ORDER BY allenvs.envvar_priority ASC;
+  `;
 
   const results = await query(sqlClientPool, sql, [envId, envId, envId, envId]);
 
-  const pendingChanges = results.map(row => {
-    return {type:environmentPendingChangeTypes.ENVVAR, details: `Variable name: ${row.envvarName} (source: ${row.envvarSource} )`, date: row.envvarUpdated};
-  });
+  const overrideMap = new Map();
+
+  const shouldDeployDBString = "deploy";
+  results.forEach((row) => {
+      if (overrideMap.has(row.envvarName)) { // Check if there is already an instance of this var
+          let other = overrideMap.get(row.envvarName);
+          if(row.envvarPriority > other.envvarPriority) { // if this is higher
+              if(row.mustDeploy === shouldDeployDBString) {
+                  // We override conventionally
+                  overrideMap.set(row.envvarName, row);
+              } else {
+                  // We override without any deployment - remove
+                  overrideMap.delete(row.envvarName);
+              }
+          }
+      } else {
+          if(row.mustDeploy === shouldDeployDBString) {
+              // First instance of a var to be deployed
+              overrideMap.set(row.envvarName, row);
+          }
+      }
+  })
+
+  const pendingChanges =  Array.from(overrideMap.values()).map((row) => {
+      return {
+        type:environmentPendingChangeTypes.ENVVAR,
+        details: `Variable name: ${row.envvarName} (source: ${row.envvarSource} )`,
```

Comment:
```suggestion
        details: `Variable name: ${row.envvarName} (source: ${row.envvarSource})`,
```

Random space after the source
```
"details": "Variable name: SSMTP_MAILHUB (source: Organization )",
```